File: dota-zone/dotazone/src/main/java/br/com/dotazone/view/fragment/HeroSkillFragment.java
package br.com.dotazone.view.fragment;

import android.os.Bundle;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import br.com.dotazone.R;
import br.com.dotazone.model.entity.Skill;
import br.com.dotazone.model.util.Utils;
import br.com.dotazone.view.components.ImageGetter;

public class HeroSkillFragment extends BaseFragment {

    private View mView;
    private Skill mSkill;
    private TextView mTextSkillName;
    private LinearLayout mLinearSkill;


    public static final HeroSkillFragment newInstance(Skill skill) {
        HeroSkillFragment f = new HeroSkillFragment();
        f.mSkill = skill;
        return f;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        mView = inflater.inflate(R.layout.hero_descripition_skill_view, container, false);

        initComponents();

        return mView;
    }

    @Override
    public void initComponents() {

        mTextSkillName = (TextView) mView.findViewById(R.id.hero_text_skill_name);
        mLinearSkill = (LinearLayout) mView.findViewById(R.id.linear_skill_description);
        mTextSkillName.setText(mSkill.getdName());

        String affects = mSkill.getAffectes().equals("") ? null : mSkill.getAffectes();
        String desc = mSkill.getDesc().equals("") ? null : mSkill.getDesc();
        String notes = mSkill.getNotes().equals("") ? null : mSkill.getNotes();
        String dmg = mSkill.getDmg().equals("") ? null : mSkill.getDmg();
        String attribs = mSkill.getAttrib().equals("") ? null : mSkill.getAttrib();
        String cmb = mSkill.getCmb().equals("") ? null : mSkill.getCmb();
        String lore = mSkill.getLore().equals("") ? null : mSkill.getLore();
        createAttributsSkill(cmb, true, 3);
        createAttributsSkill(affects, false, 20);
        createAttributsSkill(attribs, false, 20);
        createAttributsSkill(desc, false, 20);
        createAttributsSkill(notes, false, 20);
        createAttributsSkill(dmg, false, 20);
        createAttributsSkill(lore, false, 20);


    }


    private void createAttributsSkill(String attribute, boolean imageInHTML, int countLine) {

        if (attribute != null) {

            TextView textView = new TextView(getActivity());
            textView.setTextSize(13);
            textView.setTextColor(getResources().getColor(R.color.color_text_list));
            textView.setMaxLines(countLine);
            textView.setIncludeFontPadding(false);
            if (imageInHTML) {
                textView.setText(Html.fromHtml(attribute, new ImageGetter(getActivity()), null));

            } else {

                textView.setText(Html.fromHtml(attribute));
            }

            // textView.setText(((String)
            // textView.getText().toString()).replace("\n", " "));
            mLinearSkill.addView(textView);
            LinearLayout.LayoutParams paramSkill = (LinearLayout.LayoutParams) new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT,
                    LinearLayout.LayoutParams.WRAP_CONTENT);
            paramSkill.topMargin = Utils.convertDpToPixel(10, getResources());
            paramSkill.leftMargin = Utils.convertDpToPixel(10, getResources());
            paramSkill.rightMargin = Utils.convertDpToPixel(10, getResources());
            // paramSkill.setMargins(Utils.convertDpToPixel(0, getResources()),
            // Utils.convertDpToPixel(-5, getResources()),
            // Utils.convertDpToPixel(0, getResources()),
            // Utils.convertDpToPixel(0, getResources()));
            textView.setLayoutParams(paramSkill);

        }

    }
}

File: dota-zone/dotazone/src/main/java/br/com/dotazone/view/fragment/HeroSkillFragment.kt
package br.com.dotazone.view.fragment

import android.os.Bundle
import android.text.Html
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import br.com.dotazone.R
import br.com.dotazone.model.entity.Skill
import br.com.dotazone.model.util.UrlUtils
import br.com.dotazone.view.components.ImageGetter
import kotlinx.android.synthetic.main.hero_descripition_skill_view.*


class HeroSkillFragment : BaseFragment() {
	private var mView: View? = null
	private lateinit var mSkill: Skill
	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
		return inflater.inflate(R.layout.hero_descripition_skill_view, container, false)
	}

	override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
		super.onViewCreated(view, savedInstanceState)
		initComponents()
	}

	override fun initComponents() {
		heroTextSkillName.text = mSkill.getdName()
		val affects = if (mSkill.affectes == "") null else mSkill.affectes
		val desc = if (mSkill.desc == "") null else mSkill.desc
		val notes = if (mSkill.notes == "") null else mSkill.notes
		val dmg = if (mSkill.dmg == "") null else mSkill.dmg
		val attribs = if (mSkill.attrib == "") null else mSkill.attrib
		val cmb = if (mSkill.cmb == "") null else mSkill.cmb
		val lore = if (mSkill.lore == "") null else mSkill.lore
		createAttributsSkill(cmb, true, 3)
		createAttributsSkill(affects, false, 20)
		createAttributsSkill(attribs, false, 20)
		createAttributsSkill(desc, false, 20)
		createAttributsSkill(notes, false, 20)
		createAttributsSkill(dmg, false, 20)
		createAttributsSkill(lore, false, 20)
	}

	private fun createAttributsSkill(attribute: String?, imageInHTML: Boolean, countLine: Int) {
		if (attribute != null) {
			val textView = TextView(activity)
			textView.textSize = 13f
			textView.setTextColor(resources.getColor(R.color.color_text_list))
			textView.maxLines = countLine
			textView.includeFontPadding = false
			if (imageInHTML) {
				textView.text = Html.fromHtml(attribute, ImageGetter(activity), null)
			} else {
				textView.text = Html.fromHtml(attribute)
			}
			linearSkillDescription.addView(textView)
			val paramSkill = LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT,
					LinearLayout.LayoutParams.WRAP_CONTENT)
			paramSkill.topMargin = UrlUtils.convertDpToPixel(10f, resources)
			paramSkill.leftMargin = UrlUtils.convertDpToPixel(10f, resources)
			paramSkill.rightMargin = UrlUtils.convertDpToPixel(10f, resources)
			textView.layoutParams = paramSkill
		}
	}

	companion object {
		fun newInstance(skill: Skill): HeroSkillFragment {
			val f = HeroSkillFragment()
			f.mSkill = skill
			return f
		}
	}
}

